### PR TITLE
Add AHAS Sentinel external metrics adapter

### PR DESCRIPTION
**Pull request by sczyh30**
See [the document](https://github.com/sczyh30/alibaba-cloud-metrics-adapter/blob/sentinel/docs/metrics/ahas_sentinel.md) for deatils.

### CODE REVIEW

File: docs/metrics/ahas_sentinel.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+#### Global Params
+
+All metrics need the global params.
+
+| Global params       | Description              | Example            | Required | Default value |
+| ------------------- | ------------------------ | ------------------ | -------- | ------------- | 
+| `ahas.sentinel.app.name` | The name of your service in AHAS | sentinel-console | True |  |
+| `ahas.sentinel.namespace` | The namespace of your service in AHAS | staging | False | default |
+| `ahas.sentinel.stat.period` | The statistic period of request count (in second) | 5 | False | 1 |
```
Comment (ringtail) [inline docs/metrics/ahas_sentinel.md]: Is default value "1" large enough?

Comment (ringtail) [reply docs/metrics/ahas_sentinel.md]: Pls change to `ahas.sentinel.period`  and `ahas.sentinel.app`. Because the dimensions have different meanings.

**Comment by ringtail** [comment]
Pls move SDK to vendor.

File: docs/metrics/ahas_sentinel.md (hunk 1 above)
Comment (ringtail) [reply docs/metrics/ahas_sentinel.md]: Pls add some summary comment

File: pkg/metrics/ahas/ahas_sentinel.go [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	log "k8s.io/klog"
+	"k8s.io/metrics/pkg/apis/external_metrics"
+)
+
+const (
+	AHAS_SENTINEL_TOTAL_QPS = "ahas_sentinel_total_qps"
+	AHAS_SENTINEL_PASS_QPS  = "ahas_sentinel_pass_qps"
```
Comment (ringtail) [inline pkg/metrics/ahas/ahas_sentinel.go]: Pls run `make sanitize` first

File: docs/metrics/ahas_sentinel.md (hunk 1 above)
Comment (sczyh30) [reply docs/metrics/ahas_sentinel.md]: Updated, please check.

The default interval has been updated to 10s.

File: vendor/github.com/aliyun/aliyun-ahas-go-sdk/client.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
+package ahas
```
Comment (ringtail) [inline vendor/github.com/aliyun/aliyun-ahas-go-sdk/client.go]: Hmm. pls use `dep` to update vendor instead of moving source code to vendor.

**Pull request merged**